src/services/completions.ts

```diff
@@ -1997,6 +1997,9 @@ namespace ts.Completions {
 
         const validIdentifierResult: CompletionEntryDisplayNameForSymbol = { name, needsConvertPropertyAccess: false };
         if (isIdentifierText(name, target)) return validIdentifierResult;
+        if (symbol.valueDeclaration && isNamedDeclaration(symbol.valueDeclaration) && isPrivateName(symbol.valueDeclaration.name)) {

```

Comment:
> does isPrivateName imply isNamedDeclaration?

There's no implication relationship beyond that a node must have a name in order for it to have a private name.

> Should it, or should it maybe always check instead of callers checking?

The arguments to the two functions are different. If I understand correctly, the following wouldn't add value even if it were legal TS: `declare function isPrivateName(node) : node is PrivateName & node["parent"] is NamedDeclaration`



```ts
// simplified types showing relationships
type Node = NamedNode | {};
type Declaration = Node & .....
type NamedDeclaration = Declaration & { name: PrivateName | Identifier };

// real implementations
isNamedDeclaration(node) {
    return !!(node as NamedDeclaration).name; // A 'name' property should always be a DeclarationName.
}
isPrivateName(node) {
     return node.kind === SyntaxKind.PrivateName
}
```

We *could* add a function `function isPrivateNamedDeclaration(node): node is NamedDeclaration & { name: PrivateName }`, but I worry about adding helper bloat.

Reply:
can I resolve this? 
https://github.com/microsoft/TypeScript/pull/30829#discussion_r299697567